### ISSUE TITLE
Is there a way to restart the animation?

### ISSUE THREAD

**Issue by ghertyish** [luispadron/UICircularProgressRing#7]
I have a button and when it is pressed stuff on screen happens and the progress ring starts. The idea is that when the progress ring is animating the user can't press the button. I originally had this code:

    
```Swift
func animateCircularProgressView(delayTime: Int) {
        
    button.isUserInteractionEnabled = false
        
    progressRing.setProgress(value: 100, animationDuration: TimeInterval(delayTime)) {
        self.progressRing.setProgress(value: 0, animationDuration: 0, completion: nil)
        self.button.isUserInteractionEnabled = true
    }
        
}
```
which works to stop the user pressing the button whilst the timer is on but what I would rather is for when the button is clicked to reset the animation. I have this:

```Swift    
var timerIsOn: Bool = false

@IBAction func btnTapped(_ sender: AnyObject) {
    if timerIsOn {
        //playErrorSound()
        animateCircularProgressView(delayTime: time)
    } else {
         // do button stuff
    }
}

    
func animateCircularProgressView(delayTime: Int) {
        
    timerIsOn = true
        
    progressRing.setProgress(value: 100, animationDuration: TimeInterval(delayTime*2)) {
        self.progressRing.setProgress(value: 0, animationDuration: 0, completion: nil)
        self.timerIsOn = false
    }
        
}

```
But this just makes the timer disappear. Is there an easy way to just reset it whenever the button is pressed whilst it is animating?

**Comment by luispadron** [comment]
Sure, so using your code you could try something like

```swift
func animateCircularProgressView(delayTime: Int) {
    
    // This would be a class property in the view controller
    self.isProgressRingAnimating = true
    
    progressRing.setProgress(value: 100, animationDuration: TimeInterval(delayTime*2)) {
        // Remove the resetting of the progress ring from here, handle that in the button code like you wanted
        // Done animating so set it to false
        self.isProgressRingAnimating = false
    }
}
```

Then you can do 

```swift
@IBAction func btnTapped(_ sender: AnyObject) {
   // Check to see if ring is animating
    if isProgressRingAnimating {
       // You said you wanted to check to see if it was animating, and if the user pressed the button, then reset the ring
      // So to reset, just send in an animationDuration of 0
     self.progressRing.setProgress(value: 0, animationDuration: 0, completion: nil) // assuming your min value is 0
    } else {
         // do button stuff
    }
}
```

Maybe I should add a property into the `UICircularProgressRing` API which will let you know if it's animating or not, but for right now this should work.

Let me know if this is what you were after.

**Comment by ghertyish** [comment]
Hi, thanks for the reply. Sorry, I didn't mean reset as in make it 0, I meant reset the animation as in restart the animation again.

Removing the setProgress to 0 in the completion seems to stop it from disappearing which is good. I think the problem I am having now is that the timerIsOn/isProgressAnimating check is setting itself to false so after the 2nd time the button is clicked even when it is still animating it is carrying out the "// do button stuff" bit instead of reseting the animation again. I think the original animation, even though you can't see it because the progress has been set back to 0, is still going and then sets the isProgressAnimating to false. When I stick a print function in there like:

```
func animateCircularProgressView(delayTime: Int) {

    // This would be a class property in the view controller
    self.isProgressRingAnimating = true

    progressRing.setProgress(value: 100, animationDuration: TimeInterval(delayTime*2)) {
        // Remove the resetting of the progress ring from here, handle that in the button code like you wanted
        // Done animating so set it to false
        self.isProgressRingAnimating = false
       print("Done animating")
    }
}
```

When the button gets pressed again it shows "Done animating" twice. Is there a way to cancel the original animation? Sorry if this is confusing.

**Comment by luispadron** [comment]
Okay, so there was no clean way of doing this without me updating the project.

[Version 1.1.4](https://github.com/luispadron/UICircularProgressRing/releases/tag/v1.1.4)

Run `pod update` if using cocoapods.

Now every `UICircularProgressRing` has an `isAnimating` property.

This should now allow the restarting animation behavior you want:

```swift
func animateCircularProgressView(delayTime: Int) {
    progressRing.setProgress(value: 100, animationDuration: TimeInterval(delayTime*2)) {
        // Completion no longer needed, remove if you want
    }
}

@IBAction func btnTapped(_ sender: AnyObject) {
    if self.progressRing.isAnimating {
        //playErrorSound()
        // Reset the progress ring before starting it again
        self.progressRing.setProgress(value: 0, animationDuration: 0, completion: nil) // assuming your min value is 0
        animateCircularProgressView(delayTime: time)
    } else {
         // do button stuff
    }
}
```

**Comment by ghertyish** [comment]
That's awesome! Cheers mate, I really appreciate it. It's working perfectly now!
